internal/pipeline/pipeline.go

```diff
@@ -164,3 +196,82 @@ type Function struct {
 	// `ConfigMap` is a convenient way to specify a function config of kind ConfigMap.
 	ConfigMap map[string]string `yaml:"configMap,omitempty"`
 }
+
+// Validate will validate the function `f`.
+func (f *Function) Validate() error {
+	err := ValidateFunctionName(f.Image)
+	if err != nil {
+		return fmt.Errorf("function name is not valid: %w", err)
+	}
+	// configCnt is used to count the number of configs provided. More than one
+	// config is not allowed.
+	var configCnt = 0
+	if f.ConfigPath != "" {
+		configCnt++
+		if err := ValidatePath(f.ConfigPath); err != nil {
+			return fmt.Errorf("configPath is invalid: %w", err)
+		}
+	}
+	if len(f.ConfigMap) != 0 {
+		configCnt++
+	}
+	if !isNodeZero(&f.Config) {
+		configCnt++
+	}
+	if configCnt > 1 {
+		return fmt.Errorf("only zero or one config is allowed, %d provided", configCnt)
```

Comment:
General rule: Refer to path of specific fields you're validating in error messages. e.g. `spec.....ConfigMap`. Otherwise, "zero or one config is allowed" is ambigious.

Same for all other error messages.